packages/file-picker/ios/Plugin/FilePicker.swift

```diff
@@ -213,11 +213,14 @@ import MobileCoreServices
     }
 
     private func saveTemporaryFile(_ sourceUrl: URL) throws -> URL {
-        var directory = URL(fileURLWithPath: NSTemporaryDirectory())
+        let uniqueFolderName = UUID().uuidString
+        var directory = URL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent(uniqueFolderName)
         if let cachesDirectory = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first {
-            directory = cachesDirectory
+            directory = cachesDirectory.appendingPathComponent(uniqueFolderName)
         }
-        let targetUrl = directory.appendingPathComponent(sourceUrl.lastPathComponent)
+        try FileManager.default.createDirectory(at: directory, withIntermediateDirectories: true, attributes: nil)
+        let uniqueFileName = "\(directory.lastPathComponent)_\(sourceUrl.lastPathComponent)"
```

Comment:
Why is this line necessary? With this we change the original file name.

Reply:
@robingenz - Without creating a unique fileName I was noticing that the Capacitor.convertFileSrc method would still point all the files to the same instance of a file resulting in duplicates. You can see more info on the comment on the main thread that shows both the code I tried to implement at the Swift / JS layer and the result that was being output

Reply:
You mean we need a unique file path? 
And we already have a unique folder name, so this should work without renaming the file itself.

Reply:
Can you give it a try?

Reply:
@robingenz  - Changed the function to:

```
    private func saveTemporaryFile(_ sourceUrl: URL) throws -> URL {
         let uniqueFolderName = UUID().uuidString
         var directory = URL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent(uniqueFolderName)
         if let cachesDirectory = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first {
             directory = cachesDirectory.appendingPathComponent(uniqueFolderName)
         }
         try FileManager.default.createDirectory(at: directory, withIntermediateDirectories: true, attributes: nil)
         let targetUrl = directory.appendingPathComponent(sourceUrl.lastPathComponent)
         do {
             try deleteFile(targetUrl)
         }
         try FileManager.default.copyItem(at: sourceUrl, to: targetUrl)
         return targetUrl
     }
     ```
     
  And then logged out what the nativeSrc was returning after picking 3 files with the same name:
  
  ```
  ⚡️  [log] - NATIVE SRC capacitor://localhost/_capacitor_file_/private/var/mobile/Containers/Data/Application/3E00ADEE-62F0-416D-99CB-F337BB359275/tmp/GPTempDownload.jpeg
⚡️  [log] - NATIVE SRC capacitor://localhost/_capacitor_file_/private/var/mobile/Containers/Data/Application/3E00ADEE-62F0-416D-99CB-F337BB359275/tmp/GPTempDownload.jpeg
⚡️  [log] - NATIVE SRC capacitor://localhost/_capacitor_file_/private/var/mobile/Containers/Data/Application/3E00ADEE-62F0-416D-99CB-F337BB359275/tmp/GPTempDownload.jpeg

```

They are duplicating the same as they are placed in the same tmp folder with the same name

Reply:
I also just validated that the `targetUrl` is unique as well by logging it out at the native level

```
HELP file:///var/mobile/Containers/Data/Application/1DEEE4F5-97D8-4327-92F7-878B8A0BA8FF/Library/Caches/941A951B-E07D-4E1A-B1CB-BB3318FBBD76/GPTempDownload.heic
HELP file:///var/mobile/Containers/Data/Application/1DEEE4F5-97D8-4327-92F7-878B8A0BA8FF/Library/Caches/5ADCE43B-0236-4D35-847A-3A59273F6824/GPTempDownload.heic
HELP file:///var/mobile/Containers/Data/Application/1DEEE4F5-97D8-4327-92F7-878B8A0BA8FF/Library/Caches/CF10D8D2-8DE4-4C60-AABC-8BE154DEBB07/GPTempDownload.heic
```

Reply:
> I also just validated that the `targetUrl` is unique as well by logging it out at the native level
> 
> ```
> HELP file:///var/mobile/Containers/Data/Application/1DEEE4F5-97D8-4327-92F7-878B8A0BA8FF/Library/Caches/941A951B-E07D-4E1A-B1CB-BB3318FBBD76/GPTempDownload.heic
> HELP file:///var/mobile/Containers/Data/Application/1DEEE4F5-97D8-4327-92F7-878B8A0BA8FF/Library/Caches/5ADCE43B-0236-4D35-847A-3A59273F6824/GPTempDownload.heic
> HELP file:///var/mobile/Containers/Data/Application/1DEEE4F5-97D8-4327-92F7-878B8A0BA8FF/Library/Caches/CF10D8D2-8DE4-4C60-AABC-8BE154DEBB07/GPTempDownload.heic
> ```

However, with a unique `targetUrl` it should work or not? In your issue the problem was that the `path` was the same for both files. We have fixed this now with the unique folder names. The files are no longer overwritten. The thing with `nativeSrc` could be a different problem.
Or am I missing something?

Reply:
@robingenz - I think you are right but we're needing some additional logic that I am not sure where to add. When logging out the file.path at the JS layer and the native source I'm getting:

```
⚡️ [log] - FILE PATH: {"path":"file:///private/var/mobile/Containers/Data/Application/09E61D33-6E45-41D1-80CD-5AD5FA10CE5F/tmp/GPTempDownload.jpeg"}
   capacitor://localhost/_capacitor_file_/private/var/mobile/Containers/Data/Application/09E61D33-6E45-41D1-80CD-5AD5FA10CE5F/tmp/GPTempDownload.jpeg

⚡️ [log] - FILE PATH: {"path":"file:///private/var/mobile/Containers/Data/Application/09E61D33-6E45-41D1-80CD-5AD5FA10CE5F/tmp/GPTempDownload.jpeg"}
   capacitor://localhost/_capacitor_file_/private/var/mobile/Containers/Data/Application/09E61D33-6E45-41D1-80CD-5AD5FA10CE5F/tmp/GPTempDownload.jpeg

⚡️ [log] - FILE PATH: {"path":"file:///private/var/mobile/Containers/Data/Application/09E61D33-6E45-41D1-80CD-5AD5FA10CE5F/tmp/GPTempDownload.jpeg"}
   capacitor://localhost/_capacitor_file_/private/var/mobile/Containers/Data/Application/09E61D33-6E45-41D1-80CD-5AD5FA10CE5F/tmp/GPTempDownload.jpeg

```

Which tells me that despite the temporaryFile being unique there is some logic specifically around building the path that is not correct. What's happening is we are now getting 3 URLs but they are all pointing to the same file and it's still duplicating on the frontend